packages/vite/src/node/server/transformRequest.ts

```diff
@@ -258,7 +259,7 @@ async function loadAndTransform(
   }
 
   const result = ssr
-    ? await ssrTransform(code, map as SourceMap, url, {
+    ? await ssrTransform(code, map as SourceMap, url, originalCode, {
```

Comment:
Instead of passing `originalCode`, we could always generate sourcemaps. But that will cost a lot if the file is large.